src/EditorFeatures/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests_ThroughMember.cs

```diff
@@ -0,0 +1,670 @@
+        public async Task ProtectedMethodsCannotBeDelegatedThroughBaseType()
+        {
+            await TestExactActionSetOfferedAsync(
+@"abstract class Base
+{
+    protected abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Base inner;
+}", new[] { FeaturesResources.Implement_abstract_class });
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task ProtectedMethodsCanBeDelegatedThroughSameType()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    protected abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Derived inner;
+}",
+@"abstract class Base
+{
+    protected abstract void Method();
+}
+
+class Derived : Base
+{
+    Derived inner;
+
+    protected override void Method()
+    {
+        inner.Method();
+    }
+}", index: 1, title: string.Format(FeaturesResources.Implement_through_0, "inner"));
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task ProtectedInternalMethodsAreOverridden()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    protected internal abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Base inner;
+}",
+@"abstract class Base
+{
+    protected internal abstract void Method();
+}
+
+class Derived : Base
+{
+    Base inner;
+
+    protected internal override void Method()
+    {
+        inner.Method();
+    }
+}", index: 1, title: string.Format(FeaturesResources.Implement_through_0, "inner"));
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task InternalMethodsAreOverridden()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    internal abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Base inner;
+}",
+@"abstract class Base
+{
+    internal abstract void Method();
+}
+
+class Derived : Base
+{
+    Base inner;
+
+    internal override void Method()
+    {
+        inner.Method();
+    }
+}", index: 1, title: string.Format(FeaturesResources.Implement_through_0, "inner"));
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task PrivateProtectedMethodsCannotBeDelegatedThroughBaseType()
+        {
+            await TestExactActionSetOfferedAsync(
+@"abstract class Base
+{
+    private protected abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Base inner;
+}", new[] { FeaturesResources.Implement_abstract_class });
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task PrivateProtectedMethodsCanBeDelegatedThroughSameType()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    private protected abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Derived inner;
```

Comment:
Do we have tests for nullable fields? I presume we want to generate `inner?.Method();` for such cases?

Reply:
We do not have tests.  Nor am i sure that that form is actually what you want :-/   I'm ok leaving with the current behavior for now and waiting for feedback.

Note that implement-inteface-through-member doesn't do anything special here.